tools/logger/convert.c

```diff
@@ -80,6 +80,32 @@ static void log_err(FILE *out_fd, const char *fmt, ...)
 	va_end(args);
 }
 
+char *format_uid_raw(const struct sof_uuid *uid_val, int use_colors,
+		     int name_first)
+{
+	const char *name = (const char *)(uid_val + 1) + 4;
+	int off;
+	char *str;
+
+	str = calloc(1, (use_colors ? strlen(KBLU) : 0) + 1 +
+		36 + 1 + *(uint32_t *)(uid_val + 1) + 1 +
+		(use_colors ? strlen(KNRM) : 0));
```

Comment:
Maybe snprintf with a destination buffer size of 0 can be used to compute the size? Or is that usage nonstandard?

Reply:
(even if 0 is nonstandard you can use size 1 and it still works)

Reply:
Although possibly the best solution would be to provide an implementation of asprintf on systems which don't already have it. A simple one:

```
char *vasprintf(const char *format, va_list args)
{
	va_list args_copy;
	int size;
	char localbuf[1];
	char *result;

	va_copy(args_copy, args);
	size = vsnprintf(localbuf, 1, format, args_copy);
	va_end(args_copy);

	result = calloc(1, size + 1);
	if (result)
		vsnprintf(result, size + 1, format, args);
	return result;
}

char *asprintf(const char *format, ...)
{
	va_list args;
	char *result;

	va_start(args, format);
	result = vasprintf(format, args);
	va_end(args);

	return result;
}
```

Reply:
ok, introduced